### PR TITLE
Kf order status

### PR DESCRIPTION
**Pull request by TheKaterTot**
This is a fairly in-depth PR. I implemented all of the functionality for updating an order's status, including updating the totals displayed on the page. I also found opportunities to refactor our code from earlier today in order to make my code a little simpler. This includes several new methods on the order model and some corresponding changes in the views to remove some of the logic.

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -16,19 +16,34 @@ def total
   end
 
   def display_status
-    status.capitalize
+    status.capitalize if status
```
Comment (AELSchauer) [inline app/models/order.rb]: @TheKaterTot I think this should be reverted back since status should never be nil.

**Review by AELSchauer — CHANGES_REQUESTED** [review]
Hey Katy, I think the changes to create the buttons is confusing and more complex than it needs to be. It adds more lines of code to both the model and the view. To figure out what's going on, the coder would have to reference back and forth between both files multiple times to understand what's going on.

File: app/models/order.rb (hunk 1 above)
Comment (TheKaterTot) [inline app/models/order.rb:19]: This fixed a particular error I was getting. I agree that the status should not be nil, but we have to protect against edge cases.

**Comment by TheKaterTot** [comment]
I disagree. This code better follows the single responsibility model and the principles of POODR. More lines of code for the sake of readability and clarity in function isn't necessarily a bad thing.

File: app/models/order.rb (hunk 1 above)
Comment (edilenedacruz) [reply app/models/order.rb:19]: So simple but so effective. Gotta love Ruby

**Comment by edilenedacruz** [comment]
Everything looks great! Thanks for adding tests to the model.

**Pull request merged**